modules/subdyn/src/SD_FEM.f90

```diff
@@ -563,19 +589,28 @@ SUBROUTINE SD_Discrt(Init,p, ErrStat, ErrMsg)
           dx = ( x2 - x1 )/Init%NDiv
           dy = ( y2 - y1 )/Init%NDiv
           dz = ( z2 - z1 )/Init%NDiv
-          
-          d1 = TempProps(Prop1, 5)
-          t1 = TempProps(Prop1, 6)
 
-          d2 = TempProps(Prop2, 5)
-          t2 = TempProps(Prop2, 6)
-          
-          dd = ( d2 - d1 )/Init%NDiv
-          dt = ( t2 - t1 )/Init%NDiv
-          
+          if (eType == idMemberBeam) then
+
+            d1 = TempProps(Prop1, 5)
+            t1 = TempProps(Prop1, 6)
+
+            d2 = TempProps(Prop2, 5)
+            t2 = TempProps(Prop2, 6)
+            
+            dd = ( d2 - d1 )/Init%NDiv
+            dt = ( t2 - t1 )/Init%NDiv
+
              ! If both dd and dt are 0, no interpolation is needed, and we can use the same property set for new nodes/elements. otherwise we'll have to create new properties for each new node
-          CreateNewProp = .NOT. ( EqualRealNos( dd , 0.0_ReKi ) .AND.  EqualRealNos( dt , 0.0_ReKi ) )  
-          
+           
+            CreateNewProp = .NOT. ( EqualRealNos( dd , 0.0_ReKi ) .AND.  EqualRealNos( dt , 0.0_ReKi ) ) 
+
+          else
```

Comment:
we might as well be explicit here and use `else if (eType == idMemberX)` for readability

Reply:
Agreed